Source/Core/TerrainProvider.js

```diff
@@ -104,8 +104,8 @@ define([
      */
     TerrainProvider.getRegularGridIndices = function(width, height) {
         //>>includeStart('debug', pragmas.debug);
-        if (width * height > 64 * 1024) {
-            throw new DeveloperError('The total number of vertices (width * height) must be less than or equal to 65536.');
+        if (width * height >= 64 * 1024) {
```

Comment:
Why minus 1? You can fit exactly 64k vertices using unsigned short and the comparison is checking the vertex count, not the max index.